internal/gatewayapi/route.go

```diff
@@ -26,6 +26,11 @@ type RoutesTranslator interface {
 	ProcessUDPRoutes(udpRoutes []*v1alpha2.UDPRoute, gateways []*GatewayContext, resources *Resources, xdsIR XdsIRMap) []*UDPRouteContext
 }
 
+const (
```

Comment:
@lizan is a smarter way to do path segment matching in envoy rather than relying on regex matching ?

Reply:
From [GEP-1016](https://gateway-api.sigs.k8s.io/geps/gep-1016/#matcher-types) we can use suffix match (with header matcher)?